### ISSUE TITLE
request| feat(Router): Make router bind resolved data to component's @Input

### ISSUE THREAD

**Issue by lemoinem** [angular/angular#19293]
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
We currently have to rely on subscribing and keeping track of ChangeDetection manually on activated routes.

This produces bloated code and prevent a single component to be used both as a router-instanciated and selector-instanciated component. It introduces discrepancies on how to deal with a component inputs and makes for a much harder DX.

## Expected behavior
It would be great if once the ActivatedRoute has been determined, the router could try and bind the resolves' results to components `@Input()`s.

For example, if I have a Route with:
```json
{
 resolve: {
    myInput: myResolver
 }  
}
```

I could then put this in my component:
```ts
@Input() public myInput;
```

and have access to it directly in the component without having to deal with the ActivatedRoute myself.

Plus if I have a link which reuse the route, it would simply update `myInput`'s value and trigger change detection directly.

What I suggest is basically a more advanced version of #9787

I am ready to try and work on a tentative PR, but I'm at a loss as to where to start. I'm guessing we would have to add something in the `ViewRef` to make adding bindings easier... If someone could point me in the right direction or outline what would need to be done, it would help a lot.

## What is the motivation / use case for changing the behavior?
Basically, much bettter DX. More reusable components. A unified way of using dynamic/static components.
This could be reused to provide much more flexible extensions. For example, we could use a similar approach to load a MdDialog.

**Comment by Toub** [comment]
Very good idea!

From inside a component there should be difference between @Input and ActivatedRoute data.

**Comment by jasonaden** [comment]
Thanks for this one. Yes, we've been discussing something similar and there is a plan for something like this. It will likely be different than the current `resolve` since that is really only for one-time data. Since this is a feature we already have on the list, I'll close this issue for now, but good to know this would help you with your apps.

**Comment by Toub** [comment]
@jasonaden do you have a general ticket to follow the discussion about the future of angular router around resolvers?

**Comment by mackelito** [comment]
we are also looking for a smooth solution to solve this.. is there another ticket for tracking this?
